xonsh/history/json.py

```diff
@@ -131,10 +131,14 @@ def files(self, only_unlocked=False):
         This is sorted by the last closed time. Returns a list of
         (timestamp, number of cmds, file name) tuples.
         """
-        # pylint: disable=no-member
-        env = getattr(builtins, "__xonsh__.env", None)
-        if env is None:
+
+        x = getattr(builtins, "__xonsh__", None)
+        if x:
+            env = getattr(x, "env", None)
+
+        if not (x and env):
```

Comment:
```suggestion
        env = getattr(getattr(builtins, "__xonsh__", None), "env", None)
        if env is None:
```

Even if you don't go with this suggestion, I'd avoid the implicit `None` checks and use a more descriptive variable name rather than `x`.

Reply:
Or I guess we can even do:
```suggestion
        try:
            buildins.__xonsh__.env
        except AttributeError:
        	return []
```

That said we don't even use `env` here, why do we check for that??